### ISSUE TITLE
XGBoost + ECFP4 + RDKit descriptors baseline achieves MA-RAE 0.773 on competition split

### ISSUE THREAD

**Issue by srijitseal** [srijitseal/polaris#8]
## Summary

Training XGBoost on ECFP4 fingerprints + RDKit 2D descriptors with per-endpoint hyperparameter tuning (HalvingRandomSearchCV) on the original competition train/test split (5,326 / 2,282). Molecules are protonated at assay-relevant pH (7.4 for most, 6.5 for Caco-2) using dimorphite_dl before feature computation. Applying the competition's log-transform (`log10(clip(x, 1e-10) + 1)`) to all endpoints except LogD — for both training and evaluation — was critical.

### Evolution of baseline performance (MA-RAE)

| Configuration | MA-RAE |
|--------------|--------|
| RF (100 trees, default params, raw-scale training, log-eval only) | 0.892 |
| RF (500 trees, tuned params, log-scale training + eval) | 0.868 |
| XGBoost (tuned, ECFP4 + 16 physchem, log-scale train + eval) | 0.826 |
| **XGBoost (tuned, ECFP4 + full RDKit 2D, dimorphite_dl protonation)** | **0.820** |

### Per-endpoint results (final: XGBoost + full RDKit 2D + protonation)

| Endpoint | pH | R² | Spearman ρ | RAE | n_train | n_test |
|----------|-----|-----|-----------|-----|---------|--------|
| LogD | 7.4 | 0.534 | 0.719 | 0.663 | 5,039 | 2,270 |
| KSOL | 7.4 | 0.199 | 0.457 | 0.878 | 5,128 | 2,170 |
| HLM CLint | 7.4 | 0.170 | 0.546 | 0.889 | 3,759 | 782 |
| MLM CLint | 7.4 | 0.210 | 0.496 | 0.912 | 4,522 | 1,170 |
| Caco-2 Papp A>B | 6.5 | 0.105 | 0.437 | 0.903 | 2,157 | 1,616 |
| Caco-2 Efflux | 6.5 | 0.128 | 0.600 | 0.799 | 2,161 | 1,616 |
| MPPB | 7.4 | 0.140 | 0.506 | 0.913 | 1,302 | 454 |
| MBPB | 7.4 | 0.398 | 0.696 | 0.702 | 975 | 451 |
| MGMB | 7.4 | 0.347 | 0.732 | 0.780 | 222 | 209 |

### Key insight: log-transform is essential

The competition evaluates on log-transformed values for all endpoints except LogD. Training on raw-scale targets and only log-transforming at evaluation time produces poor R² because the model optimizes for raw-scale MSE, which is dominated by high-value outliers. Training on log-scale targets aligns the model's loss function with the evaluation metric.

### Effect of protonation + full descriptor suite

Adding dimorphite_dl protonation and expanding from 16 hand-picked to ~200 RDKit 2D descriptors produced mixed per-endpoint effects: Caco-2 Efflux improved from R²=-0.21 to +0.13 and MPPB from -0.07 to +0.14, but HLM CLint and Caco-2 Papp regressed. Overall MA-RAE improved marginally (0.826 → 0.820). The full descriptor suite introduces noisy features that XGBoost's built-in feature selection doesn't fully compensate for.

## Models

- XGBRegressor: `tree_method="hist"`, per-endpoint tuning via `HalvingRandomSearchCV` (50 candidates, factor=3, 3-fold CV, scoring=neg_MAE)
- Features: 2048-bit ECFP4 + full RDKit 2D descriptor suite (~200, scaled), computed on dimorphite_dl-protonated SMILES at assay pH
- Protonation: dimorphite_dl with pH ± 0.5 window, max 1 variant per molecule

## Plots

<img width="1477" height="727" alt="Image" src="https://github.com/user-attachments/assets/1c9e2dc3-efbc-4b6f-8567-125f7f9f1705" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/85870592-7286-40e8-959c-9d060b08bb7b" />
<img width="2675" height="2298" alt="Image" src="https://github.com/user-attachments/assets/28ab3809-b3f5-479a-a533-5cc39da8696e" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/87571989-9ae8-49b5-b64d-c303ab406e9f" />

## Reproduce

```bash
pixi run -e cheminformatics python notebooks/2.08-seal-baseline-performance.py
```

## Source

`notebooks/2.08-seal-baseline-performance.py`

**Comment by srijitseal** [comment]
@PatWalters 

## Bootstrap 95% confidence intervals added

Added 1,000 bootstrap resamples of test-set predictions to quantify uncertainty on the single train/test split. Each resample draws n_test samples with replacement from the (y_true, y_pred) pairs and recomputes all metrics; the 2.5th and 97.5th percentiles define the 95% CI.

### Per-endpoint metrics with 95% CIs

| Endpoint | R² [95% CI] | RAE [95% CI] | Spearman [95% CI] |
|----------|-------------|--------------|-------------------|
| LogD | 0.583 [0.551, 0.613] | 0.630 [0.608, 0.652] | 0.753 [0.732, 0.773] |
| KSOL | 0.256 [0.193, 0.310] | 0.850 [0.826, 0.876] | 0.489 [0.455, 0.520] |
| HLM CLint | 0.255 [0.171, 0.332] | 0.847 [0.800, 0.895] | 0.575 [0.524, 0.625] |
| MLM CLint | 0.177 [0.102, 0.247] | 0.925 [0.884, 0.964] | 0.452 [0.407, 0.497] |
| Caco-2 Papp A>B | 0.230 [0.183, 0.276] | 0.840 [0.812, 0.867] | 0.527 [0.492, 0.561] |
| Caco-2 Efflux | 0.191 [0.147, 0.235] | 0.775 [0.740, 0.810] | 0.646 [0.616, 0.677] |
| MPPB | 0.223 [0.094, 0.328] | 0.869 [0.812, 0.928] | 0.613 [0.550, 0.665] |
| MBPB | 0.560 [0.494, 0.615] | 0.592 [0.548, 0.636] | 0.782 [0.734, 0.821] |
| MGMB | 0.502 [0.402, 0.592] | 0.633 [0.566, 0.703] | 0.741 [0.651, 0.813] |

### CI width analysis

Endpoints with **wide CIs** (sparse test data):
- **MPPB** (n_test=454): R² CI width 0.234, RAE width 0.116, Spearman width 0.115
- **MGMB** (n_test=209): R² CI width 0.190, RAE width 0.137, Spearman width 0.162
- **HLM CLint** (n_test=782): R² CI width 0.162, RAE width 0.096, Spearman width 0.102

Endpoints with **tight CIs** (large test sets):
- **LogD** (n_test=2,270): R² CI width 0.062, RAE width 0.044, Spearman width 0.041
- **KSOL** (n_test=2,170): R² CI width 0.117, RAE width 0.050, Spearman width 0.065
- **Caco-2 Papp A>B** (n_test=1,616): R² CI width 0.092, RAE width 0.055, Spearman width 0.070

The wide CIs on MPPB and MGMB mean their point estimates are less reliable -- a 0.05 improvement on these endpoints may not be statistically meaningful on this split.

### Updated plots with error bars

<img width="1477" height="727" alt="Image" src="https://github.com/user-attachments/assets/ea5c1f00-1702-4c46-9790-7083a3308d1e" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/3c7296f6-9968-4ed5-bfa2-1db32929b18c" />
<img width="2675" height="2298" alt="Image" src="https://github.com/user-attachments/assets/cbcb82ae-ff34-40e5-8b27-38258bfac025" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/d54755bf-ff24-4708-9d96-0d32505c25d5" />

**Comment by srijitseal** [comment]
Note: We will not comment on comparisons such as halving search versus Optuna, or chiral versus non-chiral performance. We will still report error estimates, but comparing the above conditions is beyond the scope of this work. Instead, we report only the best-performing configuration for this analysis, as the primary goal of this paper is to evaluate data splitting strategies rather than optimization methods or fingerprint choices.

Chiral fingerprints are the appropriate choice in this context, given that the stereochemical information in the datasets is being verified by the authors. Additionally, Optuna was selected for hyperparameter optimization due to its ease of implementation, whereas halving search would have been impractical for the larger notebooks, which involve training on the order of 1,500 models.

**Comment by srijitseal** [comment]
## Note on GitHub issue body vs current results (2026-04-19)

Spot-checked: the local writeup at `docs/paper/issues/010-baseline-performance.md` and the per-endpoint numbers in `data/processed/2.08-seal-baseline-performance/summary_metrics.csv` agree with the values quoted in the later comments on this issue (MA-RAE 0.773 after Optuna TPE, per-endpoint R²/RAE/CIs from bootstrap, n_train/n_test = 5,326 / 2,282, etc.). Only the original issue *body* is pre-Optuna. Flagging for awareness — the comments are the source of truth, no re-analysis needed.

**Comment by srijitseal** [comment]
## CheMeleon update (2.08 baseline)

Re-ran the baseline with the CheMeleon foundation model (pre-trained BondMessagePassing weights, 2×1024 FFN, fine-tuned per endpoint, single model — no ensemble). Same competition split, same protonation, same log-transform protocol.

**Verdict: results match the XGBoost trend qualitatively, with CheMeleon strictly better on 8/9 endpoints.**

### MA-RAE comparison

| Model | MA-RAE |
|-------|--------|
| XGBoost (Optuna TPE, ECFP4 + RDKit 2D) | 0.776 |
| **CheMeleon (fine-tuned)** | **0.684** |

Δ = −0.092 (−11.9% relative).

### Per-endpoint R² and RAE

| Endpoint | R² XGB | R² CM | ΔR² | RAE XGB | RAE CM | ΔRAE |
|----------|--------|-------|-----|---------|--------|------|
| LogD | 0.592 | 0.720 | +0.128 | 0.628 | 0.489 | −0.138 |
| KSOL | 0.295 | 0.462 | +0.166 | 0.826 | 0.593 | −0.233 |
| HLM CLint | 0.231 | 0.328 | +0.096 | 0.855 | 0.799 | −0.056 |
| MLM CLint | 0.189 | 0.170 | −0.019 | 0.920 | 0.936 | +0.016 |
| Caco-2 Papp A>B | 0.192 | 0.206 | +0.014 | 0.859 | 0.830 | −0.029 |
| Caco-2 Efflux | 0.186 | 0.207 | +0.021 | 0.775 | 0.748 | −0.027 |
| MPPB | 0.223 | 0.533 | +0.310 | 0.879 | 0.674 | −0.205 |
| MBPB | 0.570 | 0.628 | +0.058 | 0.582 | 0.504 | −0.078 |
| MGMB | 0.452 | 0.463 | +0.010 | 0.659 | 0.580 | −0.079 |

Largest gains: MPPB (+0.31 R²), KSOL (+0.17), LogD (+0.13). Only regression: MLM CLint (−0.02 R²) — within the noise of a single split.

### Trend match

Yes. The baseline ranking by endpoint difficulty is preserved (LogD/MBPB/MGMB easiest; MLM CLint and Caco-2 endpoints hardest). Foundation model lifts the floor uniformly but does not change which endpoints are hard or easy.

### Source

- `data/processed/2.08-seal-baseline-performance/chemeleon/overall_metrics.csv`
- `data/processed/2.08-seal-baseline-performance/combined/metrics_comparison.csv`
- Reproduce: `pixi run -e cheminformatics python notebooks/2.08-seal-baseline-performance.py --model chemeleon`

**Comment by srijitseal** [comment]
## Refreshed numbers (XGBoost rerun + CheMeleon)

The issue body has been re-synced against current artifacts under `data/processed/2.08-seal-baseline-performance/xgboost/`. Headline drift vs the original write-up:

| | Old | Refreshed |
|---|---|---|
| MA-RAE | 0.773 | **0.776** |
| LogD R² | 0.583 | 0.592 |
| KSOL R² | 0.256 | 0.295 |
| HLM CLint R² | 0.255 | 0.231 |
| MBPB R² | 0.560 | 0.570 |
| MGMB R² | 0.502 | 0.452 |

Per-endpoint CIs and remaining R²/Spearman values were also refreshed in the body.

The CheMeleon comparison (MA-RAE 0.684, +0.13 R² on LogD, +0.31 on MPPB, only MLM regresses) stands — see #issuecomment-4376590294 for the full CheMeleon table.

**Comment by srijitseal** [comment]
**Update: CheMeleon baseline added**

CheMeleon baseline results now included alongside XGBoost. CheMeleon achieves MA-RAE 0.684 on the competition split, outperforming XGBoost (MA-RAE 0.776). Title updated to reflect both models.
